fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/context/ContainerItemContext.java

```diff
@@ -0,0 +1,192 @@
+	 * The API is queried for the current variant, if it's not blank.
+	 *
+	 * @see ItemApiLookup#find
+	 */
+	@Nullable
+	default <A> A find(ItemApiLookup<A, ContainerItemContext> lookup) {
+		return getItemVariant().isBlank() ? null : lookup.find(getItemVariant().toStack(), this);
+	}
+
+	/**
+	 * Return the current item variant of this context, that is the variant in the slot of the context.
+	 * If the result is non blank, {@link #getAmount} should be
+	 */
+	default ItemVariant getItemVariant() {
+		return getMainSlot().getResource();
+	}
+
+	/**
+	 * Return the current amount of {@link #getItemVariant()} in the slot of the context.
+	 *
+	 * @throws IllegalStateException If {@linkplain #getItemVariant() the current variant} is blank.
+	 */
+	default long getAmount() {
+		if (getItemVariant().isBlank()) {
+			throw new IllegalStateException("Amount may not be queried when the current resource is blank.");
+		}
+
+		return getMainSlot().getAmount();
+	}
+
+	/**
+	 * Try to insert some items into this context, prioritizing the main slot over the rest of the inventory.
+	 *
+	 * @see Storage#insert
+	 */
+	default long insert(ItemVariant itemVariant, long maxAmount, TransactionContext transaction) {
+		// Main slot first
+		long mainInserted = getMainSlot().insert(itemVariant, maxAmount, transaction);
+		// Overflow second
+		long overflowInserted = insertOverflow(itemVariant, maxAmount - mainInserted, transaction);
+
+		return mainInserted + overflowInserted;
+	}
+
+	/**
+	 * Try to extract some items from this context's slot.
+	 *
+	 * @see Storage#extract
+	 */
+	default long extract(ItemVariant itemVariant, long maxAmount, TransactionContext transaction) {
+		return getMainSlot().extract(itemVariant, maxAmount, transaction);
+	}
+
+	/**
+	 * Try to transform as many as possibly of {@linkplain #getItemVariant() the current variant} into another variant.
+	 * That is, extract the old variant, and insert the same amount of the new variant instead.
+	 *
+	 * @param into The variant of the items after the conversion. May not be blank.
+	 * @param maxAmount The maximum amount of items to convert. May not be negative.
+	 * @param transaction The transaction this operation is part of.
+	 * @return A nonnegative integer not greater than maxAmount: the amount that was transformed.
+	 */
+	default long transform(ItemVariant into, long maxAmount, TransactionContext transaction) {
+		StoragePreconditions.notBlankNotNegative(into, maxAmount);
+
+		try (Transaction nested = transaction.openNested()) {
+			long extracted = extract(getItemVariant(), maxAmount, nested);
+
+			if (insert(into, maxAmount, nested) == extracted) {
+				nested.commit();
+				return extracted;
+			}
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Return the main slot of this context.
+	 */
+	SingleSlotStorage<ItemVariant> getMainSlot();
+
+	/**
+	 * Try to insert overflow items into this context.
```

Comment:
this is not descriptive enough, no idea what it is supposed to do

Reply:
Better now?